src/epic-cli.cpp

```diff
@@ -179,6 +179,12 @@ int main(int argc, char** argv) {
             }
 
             case START_MINER: {
+                auto sync_completed = client.SyncCompleted();
+                if (!sync_completed || !(*sync_completed)) {
+                    std::cout << "The initial synchronization is not completed. Abort starting miner." << std::endl;
```

Comment:
if `!sync_completed`, the log should notify the failed connection between the client and the server.